src/rgw/driver/rados/rgw_user.cc

```diff
@@ -1439,6 +1439,12 @@ int RGWUser::update(const DoutPrefixProvider *dpp, RGWUserAdminOpState& op_state
     return -EINVAL;
   }
 
+  // if op_state.secondary_zone_op is true most recent keys have been fetched from master zone
+  if(op_state.is_secondary_zone_op()) {
```

Comment:
is it really necessary to track a separate "secondary zone" flag for this? couldn't we just decide based on `!op_access_keys.empty()`?